json-rpc/src/tests.rs

```diff
@@ -69,6 +80,50 @@ fn test_json_rpc_protocol() {
     assert_eq!(fetch_error(resp), -32000);
 }
 
+#[test]
+fn test_transaction_submission() {
+    let (mp_sender, mut mp_events) = channel(1);
+    let tmp_dir = TempPath::new();
+    let db = LibraDB::new(&tmp_dir);
+    let address = format!("0.0.0.0:{}", utils::get_available_port());
+    let runtime = bootstrap(address.parse().unwrap(), Arc::new(db), mp_sender);
+    let client = reqwest::blocking::Client::new();
+    let url = format!("http://{}", address);
+
+    // future that mocks shared mempool execution
+    runtime.spawn(async move {
+        let validator = MockVMValidator;
+        while let Some((txn, cb)) = mp_events.next().await {
+            let vm_status = validator.validate_transaction(txn).await.unwrap();
+            let result = if vm_status.is_some() {
+                (MempoolStatus::new(MempoolStatusCode::VmError), vm_status)
+            } else {
+                (MempoolStatus::new(MempoolStatusCode::Accepted), None)
+            };
+            cb.send(Ok(result)).unwrap();
+        }
+    });
+
+    // closure that checks transaction submission for given account
+    let txn_submission = move |sender| {
+        let keypair = compat::generate_keypair(None);
+        let txn = get_test_signed_txn(sender, 0, &keypair.0, keypair.1, None);
+        let payload = hex::encode(lcs::to_bytes(&txn).unwrap());
+        let request =
+            serde_json::json!({"jsonrpc": "2.0", "method": "submit", "params": [payload], "id": 1});
+        client.post(&url).json(&request).send().unwrap()
+    };
+
+    // check successful submission
```

Comment:
what about other status codes (e.g. mempool status code)?

Reply:
not represented in current version of API(PR). Will revisit later how to better express it with existing JSON-RPC layer